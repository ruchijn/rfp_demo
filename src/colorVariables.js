const colorOffWhite = '#EBEBEB';
const colorLightGrey = '#767676';
const colorShuttleGrey = '#54595F';
const colorCardinal = '#C12335';

export { colorOffWhite, colorLightGrey, colorShuttleGrey, colorCardinal };
