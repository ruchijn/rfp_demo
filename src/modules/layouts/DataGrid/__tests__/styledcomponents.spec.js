import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';

import { colorLightGrey, colorShuttleGrey, colorCardinal } from '../../../../colorVariables';

import {
  DataGridContainer,
  FetchDataButtonWrapper,
  FetchDataButton,
  TextWrapper,
  Text,
  TopInformationWrapper,
  TopInformationSection,
  RatesWrapper,
  RatesRow,
  RatesCol,
} from '../styledComponents';

describe('<DataGridContainer />', () => {
  it('renders correctly', () => {
    const dataGridContainerWrapperTree = renderer
      .create(<DataGridContainer>Wrapper for data grid</DataGridContainer>)
      .toJSON();
    expect(dataGridContainerWrapperTree).toMatchSnapshot();
  });
});

describe('<FetchDataButtonWrapper />', () => {
  it('renders correctly', () => {
    const fetchDataButtonWrapperTree = renderer
      .create(<FetchDataButtonWrapper>this is button wrapper</FetchDataButtonWrapper>)
      .toJSON();
    expect(fetchDataButtonWrapperTree).toMatchSnapshot();
  });
});

describe('<FetchDataButton />', () => {
  it('renders correctly', () => {
    const fetchDataButtonTree = renderer
      .create(<FetchDataButton>This is the button</FetchDataButton>)
      .toJSON();
    expect(fetchDataButtonTree).toMatchSnapshot();
  });
});

describe('<TextWrapper />', () => {
  it('renders correctly', () => {
    const textWrapperTree = renderer
      .create(<TextWrapper>this is wrapper for text</TextWrapper>)
      .toJSON();
    expect(textWrapperTree).toMatchSnapshot();
  });
});

describe('<Text />', () => {
  it('renders correctly', () => {
    const textTree = renderer.create(<Text>link</Text>).toJSON();
    expect(textTree).toMatchSnapshot();
  });

  it('renders with appropriate color for red when passed', () => {
    const redTextTree = renderer.create(<Text color="red">link</Text>).toJSON();
    expect(redTextTree).toHaveStyleRule('color', colorCardinal);
  });

  it('renders with appropriate color for shuttelGrey when passed', () => {
    const redTextTree = renderer.create(<Text color="shuttleGrey">link</Text>).toJSON();
    expect(redTextTree).toHaveStyleRule('color', colorShuttleGrey);
  });
});

describe('<TopInformationWrapper />', () => {
  it('renders correctly', () => {
    const topInformationWrapperTree = renderer
      .create(<TopInformationWrapper>this is top row wrapper</TopInformationWrapper>)
      .toJSON();
    expect(topInformationWrapperTree).toMatchSnapshot();
  });
});

describe('<TopInformationSection />', () => {
  it('renders correctly', () => {
    const topInformationSectionTree = renderer
      .create(<TopInformationSection>section for top row</TopInformationSection>)
      .toJSON();
    expect(topInformationSectionTree).toMatchSnapshot();
  });

  // eslint-disable-next-line jest/no-disabled-tests
  it.skip('renders with appropriate style', () => {
    const topInformationSectionTree = renderer
      .create(<TopInformationSection right>section for top row</TopInformationSection>)
      .toJSON();
    expect(topInformationSectionTree).toHaveStyleRule('justify-content', 'flex-end');
  });
});

describe('<RatesWrapper />', () => {
  it('renders correctly', () => {
    const ratesWrapperTree = renderer
      .create(<RatesWrapper>wrapper for data table</RatesWrapper>)
      .toJSON();
    expect(ratesWrapperTree).toMatchSnapshot();
  });
});

describe('<RatesRow />', () => {
  it('renders correctly', () => {
    const ratesRowTree = renderer.create(<RatesRow>each row of data</RatesRow>).toJSON();
    expect(ratesRowTree).toMatchSnapshot();
  });
});

describe('<RatesCol />', () => {
  it('renders correctly', () => {
    const ratesColTree = renderer.create(<RatesCol>each column of data</RatesCol>).toJSON();
    expect(ratesColTree).toMatchSnapshot();
  });

  it('renders with appropriate right border color when passed', () => {
    const ratesColTree = renderer
      .create(<RatesCol rightBorder>each column of data</RatesCol>)
      .toJSON();
    expect(ratesColTree).toHaveStyleRule('border-right', `1px solid ${colorLightGrey}`);
  });
});
