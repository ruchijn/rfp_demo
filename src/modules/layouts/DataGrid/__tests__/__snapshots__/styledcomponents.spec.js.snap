// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DataGridContainer /> renders correctly 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  max-width: 80%;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  margin: 5rem auto;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

<div
  className="c0"
>
  Wrapper for data grid
</div>
`;

exports[`<FetchDataButton /> renders correctly 1`] = `
.c0 {
  border: 1px solid #767676;
  padding: 1rem 2rem;
}

<button
  className="c0"
>
  This is the button
</button>
`;

exports[`<FetchDataButtonWrapper /> renders correctly 1`] = `
.c0 {
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
  margin-bottom: 1rem;
}

<div
  className="c0"
>
  this is button wrapper
</div>
`;

exports[`<RatesCol /> renders correctly 1`] = `
.c0 {
  -webkit-flex: 1 1 50%;
  -ms-flex: 1 1 50%;
  flex: 1 1 50%;
  padding: 0.5rem 1rem;
}

<div
  className="c0"
>
  each column of data
</div>
`;

exports[`<RatesRow /> renders correctly 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  border-bottom: 1px solid #767676;
}

<div
  className="c0"
>
  each row of data
</div>
`;

exports[`<RatesWrapper /> renders correctly 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  border: 1px solid #767676;
}

.c0:last-child {
  border-bottom: none;
}

<div
  className="c0"
>
  wrapper for data table
</div>
`;

exports[`<Text /> renders correctly 1`] = `
.c0 {
  font-size: 1.5rem;
  text-align: center;
}

<p
  className="c0"
>
  link
</p>
`;

exports[`<TextWrapper /> renders correctly 1`] = `
.c0 {
  margin-top: 1rem;
}

<div
  className="c0"
>
  this is wrapper for text
</div>
`;

exports[`<TopInformationSection /> renders correctly 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

@media screen and (min-width:992px) {
  .c0 {
    -webkit-flex: 1 1 50%;
    -ms-flex: 1 1 50%;
    flex: 1 1 50%;
  }
}

<div
  className="c0"
>
  section for top row
</div>
`;

exports[`<TopInformationWrapper /> renders correctly 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

@media screen and (min-width:992px) {
  .c0 {
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
  }
}

<div
  className="c0"
>
  this is top row wrapper
</div>
`;
